packages/ember-data/lib/serializers/rest_serializer.js

```diff
@@ -487,20 +490,22 @@ var RESTSerializer = JSONSerializer.extend({
     var payload = this.normalizePayload(rawPayload);
 
     for (var prop in payload) {
-      var typeName = this.typeForRoot(prop);
-      if (!store.modelFactoryFor(typeName, prop)) {
-        Ember.warn(this.warnMessageNoModelForKey(prop, typeName), false);
-        continue;
-      }
-      var type = store.modelFor(typeName);
-      var typeSerializer = store.serializerFor(type);
+      if (prop !== "meta") {
```

Comment:
I think mutating the payload object is ok. There are other paths through the serializer already end up deleting properties from the payload. Ideally we could clone the payload at the start of the serialization process but that ends up being really expensive with large payload in older browsers.